plugins/aws-secrets-manager/src/static/on_demand/providers/secrets/secretsManager.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { SecretsManagerServiceBase } from "./base/secretsManager.service.base";
+import { GetSecretValueCommand, SecretsManagerClient } from "@aws-sdk/client-secrets-manager";
+
+@Injectable()
+export class SecretsManagerService extends SecretsManagerServiceBase {
+  private readonly awsSecretsClient: SecretsManagerClient
+
+  constructor(protected readonly configService: ConfigService) {
+    super(configService);
+
+    this.awsSecretsClient = new SecretsManagerClient({
+      region: configService.get("AWS_REGION")
+    })
+  }
+
+  async getSecret<T>(key: string): Promise<T | null> {
+    const [secret_id, secret_name] = key.split(":")
+
+    const response = await this.awsSecretsClient.send(new GetSecretValueCommand({ SecretId: secret_id }))
+
+    const secrets_list: Record<string, string> = JSON.parse(
+      response.SecretString ? response.SecretString :
+      response.SecretBinary ? new TextDecoder().decode(response.SecretBinary) :
+      "{}"
+    )
+
+    return secrets_list[secret_name] as any
```

Comment:
It would be better for users to get a friendly error log when something goes wrong (i.e. the `secret_id` doesn't exists)

Try catching the potentially thrown exception and log the error message with the `secret_id` and `secret_name`